### ISSUE TITLE
duckdb: column indexes

### ISSUE THREAD

**Issue by paultiq** [sqlfluff/sqlfluff#7450]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

https://duckdb.org/docs/stable/sql/statements/select provides an example query that fails to parse in sqlfluff:
```sql
SELECT #1, #3 FROM tbl;
```
L:   1 | P:   8 |  PRS | Line 1, Position 8: Found unparsable section: '#1, #3'

### Expected Behaviour

Should parse

### Observed Behaviour

L:   1 | P:   8 |  PRS | Line 1, Position 8: Found unparsable section: '#1, #3'

### How to reproduce

sqlfluff lint test.sql

[tool.sqlfluff.core]
dialect = "duckdb"
templater = "jinja"

### Dialect

duckdb

### Version

4.0.0 and from main

### Configuration

[tool.sqlfluff.core]
dialect = "duckdb"
templater = "jinja"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
